File: packages/ts-sdk/src/serialize.ts
import type { WorkoutLog, WorkoutTemplate, Program, LifterProfile } from './types.js'

// ============================================
// Workout Log Serialization
// ============================================

export function serializeWorkoutLog(workout: WorkoutLog): string {
  return JSON.stringify(workout)
}

export function serializeWorkoutLogPretty(workout: WorkoutLog): string {
  return JSON.stringify(workout, null, 2)
}

// ============================================
// Workout Template Serialization
// ============================================

export function serializeWorkoutTemplate(template: WorkoutTemplate): string {
  return JSON.stringify(template)
}

export function serializeWorkoutTemplatePretty(template: WorkoutTemplate): string {
  return JSON.stringify(template, null, 2)
}

// ============================================
// Program Serialization
// ============================================

export function serializeProgram(program: Program): string {
  return JSON.stringify(program)
}

export function serializeProgramPretty(program: Program): string {
  return JSON.stringify(program, null, 2)
}

// ============================================
// Lifter Profile Serialization
// ============================================

export function serializeLifterProfile(profile: LifterProfile): string {
  return JSON.stringify(profile)
}

export function serializeLifterProfilePretty(profile: LifterProfile): string {
  return JSON.stringify(profile, null, 2)
}

File: packages/kotlin-sdk/src/main/kotlin/com/openweight/Serialize.kt
package com.openweight

import com.openweight.model.Program
import com.openweight.model.WorkoutLog
import com.openweight.model.WorkoutTemplate
import kotlinx.serialization.ExperimentalSerializationApi
import kotlinx.serialization.encodeToString
import kotlinx.serialization.json.Json

private val jsonCompact = Json { encodeDefaults = false }

@OptIn(ExperimentalSerializationApi::class)
private val jsonPretty = Json {
    encodeDefaults = false
    prettyPrint = true
    prettyPrintIndent = "  "
}

fun serializeWorkoutLog(workout: WorkoutLog): String = jsonCompact.encodeToString(workout)

fun serializeWorkoutLogPretty(workout: WorkoutLog): String = jsonPretty.encodeToString(workout)

fun serializeWorkoutTemplate(template: WorkoutTemplate): String = jsonCompact.encodeToString(template)

fun serializeWorkoutTemplatePretty(template: WorkoutTemplate): String = jsonPretty.encodeToString(template)

fun serializeProgram(program: Program): String = jsonCompact.encodeToString(program)

fun serializeProgramPretty(program: Program): String = jsonPretty.encodeToString(program)
